kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala

```diff
@@ -109,6 +111,18 @@ class KyuubiSyncThriftClient private (
           }
         } else {
           shutdownAsyncRequestExecutor()
+          Utils.tryLogNonFatalError {
+            Option(engineAliveThreadPool).foreach { pool =>
+              ThreadUtils.shutdown(pool, Duration(engineAliveProbeInterval, TimeUnit.MILLISECONDS))
+            }
+            warn(s"Removing Clients for ${_remoteSessionHandle}")
+            Seq(protocol).union(engineAliveProbeProtocol.toSeq).foreach { tProtocol =>
+              if (tProtocol.getTransport.isOpen) {
+                tProtocol.getTransport.close()
```

Comment:
may it throw exception? if yes, we need to wrap it by `Utils.tryLogNonFatalError { }` to avoid interrupting the close flow.

Reply:
HI, @pan3793 
Thank you very much for your comment.

I have slightly adjusted the code.
Thank you for your detailed explanation.

Reply:
I exchanged L117 and L118, so it's always safe to close the latter `tProtocol` even the exception thrown on closing the former.